pkg/flowaggregator/s3uploader/s3uploader_test.go

```diff
@@ -180,30 +214,25 @@ func TestFlowRecordPeriodicCommit(t *testing.T) {
 		nameRand:         nameRand,
 		clusterUUID:      fakeClusterUUID,
 	}
-	testRecord := flowrecordtesting.PrepareTestFlowRecord()
-	s3UploadProc.writeRecordToBuffer(testRecord)
-	s3UploadProc.cachedRecordCount = 1
+	mockRecord := ipfixentitiestesting.NewMockRecord(ctrl)
+	flowaggregatortesting.PrepareMockIpfixRecord(mockRecord, true)
+	s3UploadProc.CacheRecord(mockRecord)
+	assert.Equal(t, 1, s3UploadProc.cachedRecordCount)
+
 	s3UploadProc.startExportProcess()
-	// wait for ticker to tick
-	err := wait.PollImmediate(10*time.Millisecond, 1*time.Second, func() (bool, error) {
-		mockS3Uploader.testReaderMutex.Lock()
-		defer mockS3Uploader.testReaderMutex.Unlock()
-		if mockS3Uploader.testReader.Len() != 0 {
-			return true, nil
-		}
-		return false, nil
-	})
-	assert.NoError(t, err)
+	time.Sleep(100 * time.Millisecond)
```

Comment:
use `DoAndReturn` in the expectation, and use a flag (or `sync.WaitGroup`) to wait for the mock to have been called, using `assert.Eventually`.